### PR TITLE
Add force_preservation parameter to Preserver and PreserveResourceJob

### PR DESCRIPTION
**Pull request by eliotjordan**
Works towards #5642

### CODE REVIEW

File: app/jobs/preserve_resource_job.rb [hunk 1]
```diff
@@ -6,7 +6,8 @@ class PreserveResourceJob < ApplicationJob
 
   # @param id [String] resource id
   # @param lock_tokens [Array<String>] serialized lock tokens
-  def perform(id:, lock_tokens: nil)
+  # @param force_preservation [Boolean] force binary node preservation
+  def perform(id:, lock_tokens: nil, force_preservation: nil)
```
Comment (tpendragon) [inline app/jobs/preserve_resource_job.rb:10]: Suggestion: Default to false instead of nil.

File: app/services/preserver.rb [hunk 2]
```diff
@@ -2,28 +2,29 @@
 class Preserver
   # Provided as a mechanism for switching preservation strategies depending on
   # the profile. For now, there's only one, so fall down to new.
-  def self.for(change_set:, change_set_persister:, storage_adapter: nil)
+  def self.for(change_set:, change_set_persister:, storage_adapter: nil, force_preservation: nil)
```
Comment (tpendragon) [inline app/services/preserver.rb:5]: Suggestion: Default to false instead of nil

File: app/services/preserver.rb [hunk 3]
```diff
@@ -2,28 +2,29 @@
...
 
   class NullPreserver
     def self.preserve!; end
   end
 
-  attr_reader :change_set, :storage_adapter, :change_set_persister
+  attr_reader :change_set, :storage_adapter, :change_set_persister, :force_preservation
   delegate :resource, to: :change_set
-  def initialize(change_set:, storage_adapter: nil, change_set_persister:)
+  def initialize(change_set:, storage_adapter: nil, change_set_persister:, force_preservation: nil)
```
Comment (tpendragon) [inline app/services/preserver.rb:16]: Suggestion: Default to false instead of nil, then you don't have to do the || below

**Review by tpendragon — CHANGES_REQUESTED** [review]
Just a suggestion for defaulting to false for the param. In my experience, things can get messy the more we use nil as a stand-in for false.

File: app/jobs/preserve_resource_job.rb (hunk 1 above)
Comment (eliotjordan) [reply app/jobs/preserve_resource_job.rb:10]: Good idea. 👍

**Review by tpendragon — APPROVED** [review]

**Pull request merged**